### ISSUE TITLE
Learning rate scheduling with DistributedLearner

### ISSUE THREAD

**Issue by 1ytic** [Microsoft/CNTK#1915]
The number of seen samples in distributed and local learner is different. If set full `epoch_size` in `learning_rate_schedule`, then scheduling will be wrong.

**Comment by KeDengMS** [comment]
Number of seen samples in distributed learner and local learner should be the same if using the same minibatch size from reader. Note there may be differences between samples seen vs. minibatch size when sample count approach epoch_size boundary. If this is not the case, could you provide a repro for the issue?

**Comment by 1ytic** [comment]
Sorry, to lazy to reproduce, but if I remember correctly, the issue relates to the [line](https://github.com/Microsoft/CNTK/blob/148a049f1f91579920e2518cdc1bf6631bf5dad7/Source/CNTKv2LibraryDll/API/CNTKLibrary.h#L4692) where DistributedLearner uses the local learner with its own m_sampleCount. 
For a repro you can take any example with, for example [TrainResNet_CIFAR10_Distributed.py](https://github.com/Microsoft/CNTK/blob/master/Examples/Image/Classification/ResNet/Python/TrainResNet_CIFAR10_Distributed.py), and run it with block_momentum_distributed_learner.

**Comment by KeDengMS** [comment]
Yes you are right that in block momentum, the distributed learner sample count may be different from local sample count. We have a sweep based learning rate schedule now, and you can try set epoch_size to FULL_DATA_SWEEP instead of using sample count. sweepEnd signal would be generated by reader to make sure learner scheduler was updated correctly.

**Comment by 1ytic** [comment]
Thank you. I will close the issue. Anyway, I am prefer BrainScript.
